Description: fix bashisms
Bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=772233
Author: Raphael Geissert <geissert@debian.org>
Forwarded: https://gnunet.org/bugs/view.php?id=3588

--- a/src/gns/gnunet-gns-proxy-setup-ca.in
+++ b/src/gns/gnunet-gns-proxy-setup-ca.in
@@ -229,7 +229,7 @@
                 verbosity=1
                 ;;
             c)
-                options="$options -c $OPTARG"
+                options+="-c $OPTARG"
                 infomsg "Using configuration file $OPTARG"
                 GNUNET_CONFIG_FILE=${OPTARG}
                 ;;
