Description: fix bashisms
Bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=772233
Author: Raphael Geissert <geissert@debian.org>
Forwarded: https://gnunet.org/bugs/view.php?id=3588

--- a/src/gns/gnunet-gns-proxy-setup-ca.in
+++ b/src/gns/gnunet-gns-proxy-setup-ca.in
@@ -15,7 +15,7 @@
 while getopts "c:" opt; do
   case $opt in
     c)
-      options="$options -c $OPTARG"
+      options+="-c $OPTARG"
       ;;
     \?)
       echo "Invalid option: -$OPTARG" >&2
