--- a/configure.ac
+++ b/configure.ac
@@ -91,7 +91,7 @@
      UNIXONLY="#"
      AC_PATH_XTRA
      ;;
-freebsd*)
+*freebsd*)
      AC_DEFINE_UNQUOTED(SOMEBSD,1,[This is a BSD system])
      AC_DEFINE_UNQUOTED(FREEBSD,1,[This is a FreeBSD system])
      CFLAGS="-D_THREAD_SAFE $CFLAGS"
@@ -100,7 +100,7 @@
      DLLDIR=lib
      UNIXONLY="#"
      ;;
-openbsd*)
+*openbsd*)
      AC_DEFINE_UNQUOTED(SOMEBSD,1,[This is a BSD system])
      AC_DEFINE_UNQUOTED(OPENBSD,1,[This is an OpenBSD system])
      LIBS=`echo $LIBS | sed -e "s/-ldl//"`
@@ -109,7 +109,7 @@
      DLLDIR=lib
      UNIXONLY="#"
      ;;
-netbsd*)
+*netbsd*)
      AC_DEFINE_UNQUOTED(SOMEBSD,1,[This is a BSD system])
      AC_DEFINE_UNQUOTED(NETBSD,1,[This is a NetBSD system])
      LIBPREFIX=
