--- a/m4/ax_create_pkgconfig_info.m4
+++ b/m4/ax_create_pkgconfig_info.m4
@@ -12,8 +12,8 @@
 #     $2 = (empty)
 #     $3 = $PACKAGE_LIBS $LIBS (as set at that point in configure.ac)
 #     $4 = $PACKAGE_SUMMARY (or $1 Library)
-#     $5 = $CPPFLAGS $PACKAGE_CFLAGS (as set at the point in configure.ac)
-#     $6 = $LDFLAGS $PACKAGE_LDFLAGS (as set at the point in configure.ac)
+#     $5 = $PACKAGE_CFLAGS (as set at the point in configure.ac)
+#     $6 = $PACKAGE_LDFLAGS (as set at the point in configure.ac)
 #
 #     PACKAGE_NAME defaults to $PACKAGE if not set.
 #     PACKAGE_LIBS defaults to -l$PACKAGE_NAME if not set.
@@ -152,13 +152,13 @@
 AC_MSG_RESULT($PKGCONFIG_libs)
 
 AC_MSG_CHECKING(our pkgconfig cppflags)
-PKGCONFIG_cppflags="ifelse($5,,$CPPFLAGS $PACKAGE_CFLAGS,$5)"
+PKGCONFIG_cppflags="ifelse($5,,$PACKAGE_CFLAGS,$5)"
 PKGCONFIG_cppflags=`eval echo "$PKGCONFIG_cppflags"`
 PKGCONFIG_cppflags=`eval echo "$PKGCONFIG_cppflags"`
 AC_MSG_RESULT($PKGCONFIG_cppflags)
 
 AC_MSG_CHECKING(our pkgconfig ldflags)
-PKGCONFIG_ldflags="ifelse($6,,$LDFLAGS $PACKAGE_LDFLAGS,$5)"
+PKGCONFIG_ldflags="ifelse($6,,$PACKAGE_LDFLAGS,$6)"
 PKGCONFIG_ldflags=`eval echo "$PKGCONFIG_ldflags"`
 PKGCONFIG_ldflags=`eval echo "$PKGCONFIG_ldflags"`
 AC_MSG_RESULT($PKGCONFIG_ldflags)
@@ -185,7 +185,7 @@
 v="$ac_top_srcdir" ;
 test ".$v" != "." || v="$ax_spec_dir"
 test ".$v" != "." || v="$srcdir"
-case "$v" in /*) PKG_CONFIG_src_headers="" ;; esac
+case "$v" in /*) PKGCONFIG_src_headers="" ;; esac
 PKGCONFIG_src_headers=`AS_DIRNAME("$PKGCONFIG_src_headers/$v/x")`
 test ! -d $PKGCONFIG_src_headers/incl[]ude || \
 PKGCONFIG_src_headers="$PKGCONFIG_src_headers/incl[]ude"
@@ -210,6 +210,8 @@
 exec_prefix=@exec_prefix@
 bindir=@bindir@
 libdir=@libdir@
+datarootdir=@datarootdir@
+datadir=@datadir@
 sysconfdir=@sysconfdir@
 includedir=@includedir@
 package=@PACKAGE@
@@ -229,6 +231,8 @@
 s|@exec_prefix@|${pkgconfig_execprefix}|
 s|@bindir@|${pkgconfig_bindir}|
 s|@libdir@|${pkgconfig_libdir}|
+s|@datarootdir@|${pkgconfig_datarootdir}|
+s|@datadir@|${pkgconfig_datadir}|
 s|@sysconfdir@|${pkgconfig_sysconfdir}|
 s|@includedir@|${pkgconfig_includedir}|
 s|@suffix@|${pkgconfig_suffix}|
@@ -252,6 +256,8 @@
 s|@exec_prefix@|${pkgconfig_execprefix}|
 s|@bindir@|${pkgconfig_bindir}|
 s|@libdir@|${pkgconfig_src_libdir}|
+s|@datarootdir@|${pkgconfig_datarootdir}|
+s|@datadir@|${pkgconfig_datadir}|
 s|@sysconfdir@|${pkgconfig_sysconfdir}|
 s|@includedir@|${pkgconfig_src_headers}|
 s|@suffix@|${pkgconfig_suffix}|
@@ -279,6 +285,8 @@
 s|@exec_prefix@|\"${pkgconfig_execprefix}\"|
 s|@bindir@|\"${pkgconfig_bindir}\"|
 s|@libdir@|\"${pkgconfig_src_libdir}\"|
+s|@datarootdir@|\"${pkgconfig_datarootdir}\"|
+s|@datadir@|\"${pkgconfig_datadir}\"|
 s|@sysconfdir@|\"${pkgconfig_sysconfdir}\"|
 s|@includedir@|\"${pkgconfig_src_headers}\"|
 s|@suffix@|\"${pkgconfig_suffix}\"|
@@ -316,6 +324,8 @@
 pkgconfig_bindir='$bindir'
 pkgconfig_libdir='$libdir'
 pkgconfig_includedir='$includedir'
+pkgconfig_datarootdir='$datarootdir'
+pkgconfig_datadir='$datadir'
 pkgconfig_sysconfdir='$sysconfdir'
 pkgconfig_suffix='$ax_create_pkgconfig_suffix'
 pkgconfig_package='$PACKAGE_NAME'
